Automattic/wp-calypso
# Me: Refactor and modernize NextSteps

This PR refactors and modernizes the `NextSteps` component. There should be no visual, functional or behavioral changes. Key points in this refactor are:

* Remove the `trophiesData` usage from `NextSteps`, as it wasn't used at all for some reason.
* Delete `lib/trophies-data` as it's no longer used.
* Update `NextSteps` to use Redux analytics actions instead of `lib/analytics` directly.
* Convert `NextSteps` to an ES6 class.
* Remove `sites-list` usage in favor of 2 new selectors that we introduce:
  * `getNewestSite` - retrieves the arguably newest site of this user
  * `hasPurchasedAPlan` - returns whether the user has at least one paid plan

Additionally, many minor modernizations and enhancements were done, including:

* Clean up a bunch of obsolete code in `NextSteps`
* Get rid of `var` usages
* Use arrow functions and get rid of unnecessary `bind` calls
* Take advantage of prop destructuring
* Use `classNames` instead of raw className string building
* Rename and reorder methods accordingly
* Reorder imports alphabetically 🔤 

To test:
* Checkout this branch.
* Go to http://calypso.localhost:3000/me/next and verify everything works like it did before.
* Go to http://calypso.localhost:3000/me/next/welcome and verify everything works like it did before.
* Test both of the above pages with a clean Redux state.
